test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESCloudBasedRepositoryIntegTestCase.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.repositories.blobstore;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import org.apache.http.HttpStatus;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.SuppressForbidden;
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.network.InetAddresses;
+import org.elasticsearch.mocksocket.MockHttpServer;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Integration tests for {@link BlobStoreRepository} implementations that relies on cloud-based
+ * services. This abstract class is responsible of starting and stopping an internal HTTP server
+ * which is used to simulate an external storage service like S3.
+ */
+@SuppressForbidden(reason = "this test uses a HttpServer to emulate a cloud-based storage service")
+public abstract class ESCloudBasedRepositoryIntegTestCase extends ESBlobStoreRepositoryIntegTestCase {
```

Comment:
NIT: maybe easier to understand what's going on here if we called this `ESMockAPIBasedRepositoryIntegTestCase` or so? We have a bunch of different test types now running against all kinds of mocks but also real APIs => would be nice if naming suggested this is a mock-based and not a real cloud API based tests? :)

Reply:
Right, now reading it again `CloudBased` is indeed misleading. I'll borrow `ESMockAPIBasedRepositoryIntegTestCase` as I don't have a better suggestion :)